modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GeneralizedPlaneStrain/GeneralizedPlaneStrainAction.md

```diff
@@ -1,15 +1,19 @@
-<!-- MOOSE Documentation Stub: Remove this when content is added. -->
-
 # GeneralizedPlaneStrainAction
 
-!alert construction title=Undocumented Class
-The GeneralizedPlaneStrainAction has not been documented, if you would like to contribute to MOOSE by writing
-documentation, please see [/generate.md]. The content contained on this page explains the typical
-documentation associated with an action; however, what is contained is ultimately determined by what
-is necessary to make the documentation clear for users.
-
 !syntax description /Modules/TensorMechanics/GeneralizedPlaneStrain/GeneralizedPlaneStrainAction
 
-!syntax parameters /Modules/TensorMechanics/GeneralizedPlaneStrain/GeneralizedPlaneStrainAction
+This action simplifies the input file syntax for creating a generalized plane strain model. It creates following MOOSE objects (i.e. Kernel, ScalarKernel and UserObject) related to the out-of-plane scalar variable.
+
+### Kernel
+
+- [Generalized Plane Strain Off-diagonal Kernel](/GeneralizedPlaneStrainOffDiag.md): in-plane displacement variables and scalar out-of-plane strain coupling
 
-!bibtex bibliography
+### ScalarKernel
+
+- [Generalized Plane Strain ScalarKernel](/GeneralizedPlaneStrain.md): out-of-plane equilibrium condition
+
+### UserObject
+
+- [Generalized Plane Strain UserObject](/GeneralizedPlaneStrainUserObject.md): residual and diagonal Jacobian calculation for scalar out-of-plane strain variable
+
```

Comment:
Would you give equations for each of these three classes that are created, especially to explain the difference between a normal Kernel and a Scalar Kernel?